src/utils/DateUtils.ts

```diff
@@ -29,111 +130,174 @@ class DefaultDateToStringOptions extends Options implements IDateToStringOptions
 }
 
+/**
+ * This class modifies the received Date so that it is rendered the way the options specify.
+ */
 export class DateUtils {
-  static convertFromJsonDateIfNeeded(date: string): Date;
-  static convertFromJsonDateIfNeeded(date: number): Date;
-  static convertFromJsonDateIfNeeded(date: Date): Date;
+
   static convertFromJsonDateIfNeeded(date: any): Date {
+    return DateUtils.convertToStandardDate(date);
+  }
+
+  static convertToStandardDate(date: string): Date;
+  static convertToStandardDate(date: number): Date;
+  static convertToStandardDate(date: Date): Date;
+  /**
+   * Receives a date of any type and converts it to a standard Date object.
+   * @param date can be a string, a number or a Date object. It can also be any other type, but the result will be
+   * 'undefined' if it is not recognized as a date.
+   * @returns {any} the Date, or undefined it it wasn't recognized as a valid date.
+   */
+  static convertToStandardDate(date: any): Date {
     if (_.isDate(date)) {
-      return date;
+      return moment(date).toDate();
     } else if (date !== null && !isNaN(Number(date))) {
-      return new Date(Number(date));
+      return moment(Number(date)).toDate();
     } else if (_.isString(date)) {
-      return new Date(<string>date.replace('@', ' '));
+      /*
+      If the input is a string recognized as a Date, parse it using momentJS.
+      This array specifies the possible string formats the moment will have to parse. It will be either the standard
+      ISO_8601 or the format our index uses to store dates ('YYYY/MM/DD@HH:mm:ssZ').
+      */
+      const dateMoment = moment(date, ['YYYY/MM/DD@HH:mm:ssZ', moment.ISO_8601]);
+      return dateMoment.toDate();
     } else {
       return undefined;
     }
   }
 
-  static dateForQuery(date: Date): string {
-    return date.getFullYear() + '/' + DateUtils.padNumber((date.getMonth() + 1).toString()) + '/' + DateUtils.padNumber(date.getDate().toString());
-  }
+  public static setLocale(): void {
+    let currentLocale = String['locale'];
 
-  private static padNumber(num: string, length: number = 2): string {
-    while (num.length < length) {
-      num = '0' + num;
+    // Our cultures.js directory contains 'no' which is the equivalent to 'nn' for momentJS
+    if (currentLocale == 'no') {
+      currentLocale = 'nn';
+    } else if (currentLocale == 'es-ES') {
+      // Our cultures.js directory contains 'es-es' which is the equivalent to 'es' for momentJS
+      currentLocale = 'es';
     }
-    return num;
-  }
 
-  static keepOnlyDatePart(date: Date): Date {
-    return new Date(date.getFullYear(), date.getMonth(), date.getDate());
+    moment.locale(currentLocale);
+
+    moment.updateLocale(currentLocale, {
+      calendar: {
+        lastDay: `[${l('Yesterday')}]`,
+        sameDay: `[${l('Today')}]`,
+        nextDay: `[${l('Tomorrow')}]`
+      }
+    });
   }
 
-  static offsetDateByDays(date: Date, offset: number): Date {
```

Comment:
This is a public method. I don't know if there was a discussion about this but be aware that this is a breaking change.